packages/element-lib/src/adapters/database/ElementRXDBAdapter.spec.js

```diff
@@ -0,0 +1,108 @@
+const ElementRXDBAdapter = require('./ElementRXDBAdapter');
+
+let db;
+const dbName = 'elementrxdb';
+
+beforeAll(async () => {
+  db = new ElementRXDBAdapter({
+    name: dbName,
+  });
+});
+
+afterAll(async () => {
+  await db.close();
+});
+
+describe('ElementPouchDBAdapter', () => {
+  const id1 = 'id1';
+  const id2 = 'id2';
+  const id3 = 'id3';
+  const type1 = 'type1';
+  const type2 = 'type2';
+  const type3 = 'type3';
+  const anchorFileHash1 = 'anchorFileHash1';
+  const anchorFileHash2 = 'anchorFileHash2';
+
+  describe('constructor', () => {
+    it('should set dbName', () => {
+      expect(db.name).toBe(dbName);
+    });
+
+    it('should set adapter', () => {
+      expect(db.adapter).toBeDefined();
+    });
+  });
+
+  describe('write', () => {
+    it('should create a record', async () => {
+      const record = await db.write(id1, {
+        anchorFileHash: anchorFileHash1,
+      });
+      expect(record.id).toBe(id1);
+      expect(record.anchorFileHash).toBe(anchorFileHash1);
+    });
+
+    it('overwrite the record if same id is specified', async () => {
+      const record = await db.write(id1, {
+        anchorFileHash: anchorFileHash2,
+        type: type1,
+      });
+      expect(record.id).toBe(id1);
+      expect(record.anchorFileHash).toBe(anchorFileHash2);
+      expect(record.type).toBe(type1);
+    });
+  });
+
+  describe('read', () => {
+    it('should return the record if it exists', async () => {
+      const record = await db.read(id1);
+      expect(record.id).toBe(id1);
+      expect(record.anchorFileHash).toBe(anchorFileHash2);
+      expect(record.type).toBe(type1);
+    });
+
+    it('should return null if record does not exist', async () => {
+      const record = await db.read(id2);
+      expect(record).toEqual([]);
```

Comment:
this tests returns `[]` either the test title should change or the return value should change.

Reply:
Ah yes you're right. It was null initially but then I changed it temporarily to [] to be consistent with the other adapter. This will be fixed in https://github.com/decentralized-identity/element/issues/49 but I will change the text for now